pkg/nsxt/nsxt_connector.go

```diff
@@ -0,0 +1,175 @@
+			}
+
+			apiToken, err := getAPIToken(nsxtConfig.VMCAuthHost, nsxtConfig.VMCAccessToken)
+			if err != nil {
+				return nil, err
+			}
+
+			securityCtx.SetProperty(security.AUTHENTICATION_SCHEME_ID, security.OAUTH_SCHEME_ID)
+			securityCtx.SetProperty(security.ACCESS_TOKEN, apiToken)
+		} else {
+			if nsxtConfig.User == "" {
+				return nil, fmt.Errorf("username must be provided")
+			}
+
+			if nsxtConfig.Password == "" {
+				return nil, fmt.Errorf("password must be provided")
+			}
+
+			securityCtx.SetProperty(security.AUTHENTICATION_SCHEME_ID, security.USER_PASSWORD_SCHEME_ID)
+			securityCtx.SetProperty(security.USER_KEY, nsxtConfig.User)
+			securityCtx.SetProperty(security.PASSWORD_KEY, nsxtConfig.Password)
+		}
+	}
+
+	tlsConfig, err := getConnectorTLSConfig(nsxtConfig.InsecureFlag, nsxtConfig.ClientAuthCertFile, nsxtConfig.ClientAuthKeyFile, nsxtConfig.CAFile)
+	if err != nil {
+		return nil, err
+	}
+	httpClient := http.Client{
+		Transport: &http.Transport{
+			Proxy:           http.ProxyFromEnvironment,
+			TLSClientConfig: tlsConfig,
+		},
+	}
+
+	connector := client.NewRestConnector(url, httpClient)
+	connector.SetSecurityContext(securityCtx)
+	if nsxtConfig.RemoteAuth {
+		connector.AddRequestProcessor(newRemoteBasicAuthHeaderProcessor())
+	}
+
+	return connector, nil
+}
+
+// getConnectorTLSConfig loads certificates to build TLS configuration
+func getConnectorTLSConfig(insecure bool, clientCertFile string, clientKeyFile string, caFile string) (*tls.Config, error) {
+	tlsConfig := tls.Config{InsecureSkipVerify: insecure}
+
+	if len(clientCertFile) > 0 {
+		if len(clientKeyFile) == 0 {
+			return nil, fmt.Errorf("Please provide key file for client certificate")
+		}
+
+		cert, err := tls.LoadX509KeyPair(clientCertFile, clientKeyFile)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to load client cert/key pair: %v", err)
+		}
+
+		tlsConfig.Certificates = []tls.Certificate{cert}
+	}
+
+	if len(caFile) > 0 {
+		caCert, err := ioutil.ReadFile(caFile)
+		if err != nil {
+			return nil, err
+		}
+
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+
+		tlsConfig.RootCAs = caCertPool
+	}
+
+	tlsConfig.BuildNameToCertificate()
+
+	return &tlsConfig, nil
+}
+
+type jwtToken struct {
+	IDToken      string `json:"id_token"`
+	TokenType    string `json:"token_type"`
+	ExpiresIn    string `json:"expires_in"`
+	Scope        string `json:"scope"`
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+}
+
+// getAPIToken gets JWT access token
+func getAPIToken(vmcAuthHost string, vmcAccessToken string) (string, error) {
+
+	payload := strings.NewReader("refresh_token=" + vmcAccessToken)
+	req, _ := http.NewRequest("POST", "https://"+vmcAuthHost, payload)
+
+	req.Header.Add("content-type", "application/x-www-form-urlencoded")
+	res, err := http.DefaultClient.Do(req)
+
+	if err != nil {
+		return "", err
+	}
+
+	if res.StatusCode != 200 {
```

Comment:
I know this was just copied from the old package but this should use `http.StatusOK` constant. Follow-up PR to fix this is fine.